nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -2087,18 +2087,17 @@ public void setRetainUntil(DocumentRef docRef, Calendar retainUntil, String comm
         checkPermission(doc, SET_RETENTION);
         Map<String, Serializable> options = new HashMap<>();
         options.put(CoreEventConstants.RETAIN_UNTIL, retainUntil);
-        String commentStart = retainUntil == null ? null : retainUntil.toInstant().toString();
-        if (comment == null) {
-            comment = commentStart;
-        } else if (commentStart != null) {
-            comment = commentStart + " " + comment;
-        }
-        options.put("comment", comment);
+        options.put("comment", retainUntil == null ? "" : retainUntil.toInstant().toString());
```

Comment:
Since this will appear on client side, my only concern is the date format that may not match what is expected by the user.
Also, for the UI this is just a string to be rendered, no further logic since no further information is provided (e.g. type or format).

Reply:
Not sure to understand the point but:
- the expiration date (when known) is mandatory in the setRetainUntil  audit log
- for setRetainUntil audit log, the only possible comment is the expiration date
- for setRetainUntil audit log, the expiration date can be updated (when extending the retention)
- the only case where the comment part can contain a text from the user is when applying a legal hold

Reply:
@jaubenque well at the API level we allow a comment at the time we set a retention, it could be used to propagate a user-level message about the reason why retention is set/extended. Even if it's not the case today I think we should leave the possibility open so not "burn" this field with a date which is already available (technically) elsewhere (in the options). The comment field is really reserved for user-level messages.

Reply:
Also, as already reported, on client side that comment is just text, we don't have further information about its nature (i.e. if it represents a date, a document reference, or so).
@jaubenque given these infos, would it make sense to review / rework those audit entries requirements?

Reply:
I understand the issue but displaying the date into the audit log is a mandatory requirement for retention. In some situations, as a litigation, you can have auditor people who will inspect all the events occuring on a given document.
If we only displayed the date in the header, you won't be able to get the full history regarding the retention period but only the current status.

Reply:
IMHO the date should go in the extended audit data so it does fulfil the requirement of having it in the audit log without abusing the human readable / unformatted "comment" field. The audit data is by default persisted in ES afaik and there are plenty of better ways to analyse this data than through the Web UI admin center. If we want to have that available in WebUI anyway I propose we make the entended audit entry data available in the UI somehow (but not in a way that requires parsing it, probably some sort of JSON on mouse hover)

Reply:
see [comment](https://github.com/nuxeo/nuxeo/pull/3849#issuecomment-603932692) below